torch/_dynamo/variables/user_defined.py

```diff
@@ -2073,7 +2073,7 @@ class MutableMappingVariable(UserDefinedObjectVariable):
     def __init__(self, value, **kwargs):
         super().__init__(value, **kwargs)
         self.generic_dict_vt = variables.ConstDictVariable({})
-        self.mutation_type = AttributeMutationExisting()
+        self.mutation_type = AttributeMutationNew()
```

Comment:
I don't understand this one

Reply:
From the source code:

```
class AttributeMutationExisting(AttributeMutation):
    """
    This case of VariableTracker.mutation_type marker indicates
    1. ...
    2. The value exists before Dynamo tracing started.

class AttributeMutationNew(AttributeMutation):
    """
    This case of VariableTracker.mutation_type marker indicates
    1. ...
    2. The value is created by the bytecode Dynamo is tracing through.
```

The value has no source and cannot be AttributeMutationExisting.

Reply:
You're right. So, the mutation type depends on whether MutableMapping has a source or not.